README.md

```diff
@@ -1 +1,50 @@
-# echo-server
\ No newline at end of file
+# Echo Server
+
+This project is built in [C#][3] and [.NET][2].
+
+.NET is a free, cross-platform, open source developer platform for building many different types of applications.
+
+C# is a simple, modern, object-oriented, and type-safe programming language.
+
+## Installation:
+
+1. To download .NET Core, visit [this][4] website.
+2. This project uses .NET version `6.0.202`. You can run the command `dotnet --version` to determine which version you have installed, or visit the .NET docs to install the correct version.
+
+## Testing:
+
+This project uses xUnit.net for testing. Visit [this][5] website for more information and how to use xUnit.net.
+
+## [Description:][1]
+
+### Functional Requirements
+
+A user should be able to interact with the echo server as follows:
+
+- When a client sends a message to the server, the server sends a response back to the client containing the original message.
+- A client can send multiple messages to the server and get the echoed response back each time.
+- Multiple clients can send messages to server and get back their proper responses.
+
+### Implementation Requirements
+
+- The server should establish a socket connection with the client using a low-level socket library. The goal of this exercise is to work with sockets directly.
+- The server should accept and return streams of data rather than raw strings.
+- The echo server should be covered by a robust suite of tests.
+
+## To Run the App:
+
+Ensure you are in the correct folder - `/echo-server/EchoServer`
+
+Type `dotnet run` in your command line prompt
+
+## To Run the Tests:
```

Comment:
Wondering if it would be more intuitive to have this as a subsection of `Testing`